libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Angular imports
+ */
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+
+/**
+ * 3rd-party imports
+ */
+
+
+/**
+ * Internal imports
+ */
+import {
+  EMPTY_LABEL_ERROR,
+} from '../button.component';
+import { DsfrButtonComponent } from '../button.component';
+import { TestHostComponent } from './test-host.component';
+import { DsfrButtonHarness } from './button.harness';
+import { ElementAlignment } from '@betagouv/ngx-dsfr';
+
+describe('DsfrButtonComponent', () => {
+  let fixture: ComponentFixture<TestHostComponent>;
+  let testHost: TestHostComponent;
+  let componentUnderTest: DsfrButtonComponent;
+  let harnessLoader: HarnessLoader;
+  let dsfrButtonHarness: DsfrButtonHarness;
+
+  const testLabel = 'Test Label';
+  const testTitle = 'Test Title';
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [
+        DsfrButtonComponent,
+        TestHostComponent,
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    testHost = fixture.componentInstance;
+    componentUnderTest = fixture.debugElement.query(
+      By.directive(DsfrButtonComponent)
+    ).componentInstance;
+
+    /*
+     * We're retrieving a HarnessLoader HERE to be able to get
+     * the Test Harness for the DsfrButtonComponent LATER.
+     *
+     * We're NOT doing both HERE since getHarness() triggers
+     * indirectly the Change Detection mechanism and thus,
+     * Data Binding and ngOnInit()...
+     */
+    harnessLoader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should be created', () => {
+    expect(componentUnderTest).toBeTruthy();
+  });
+
+  it('should throw an error when no label is provided', async () => {
+    try {
+      fixture.detectChanges();
+      throw 'It should have thrown an error about "label"';
+    } catch (error) {
+      expect(error).toBe(EMPTY_LABEL_ERROR);
+    }
+  });
+
+  describe(' all required properties are provided, ', () => {
+    beforeEach(() => {
+      testHost.testLabel = testLabel;
+    });
+
+    describe(' and inline is set to false, ', () => {
```

Comment:
I don't see any reason to add a level of separation per use-case for this Component. So we should be able to make it with only 1 nested `describe`, the one for ' all required properties are provided, '.

Therefore, I suggest you merge the `beforeEach` of lines 73 and 79.

For the rest of the test cases, you can pretty much follow what I did for the `DsfrLinkComponent` and adapt. For instance, there should be a unit test testing what happens when a consumer wants to use icons with this Component, etc.